src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.cs

```diff
@@ -31,38 +31,39 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         ? new CompilationData((CSharpCompilation)compilation)
                         : null);
 
-            IncrementalValuesProvider<BinderInvocation> inputCalls = context.SyntaxProvider
+            IncrementalValueProvider<(GenerationSpec?, ImmutableEquatableArray<DiagnosticInfo>)> genSpec = context.SyntaxProvider
                 .CreateSyntaxProvider(
                     (node, _) => node is InvocationExpressionSyntax invocation,
-                    BinderInvocation.Create)
-                .Where(operation => operation is not null);
-
-            IncrementalValueProvider<(CompilationData?, ImmutableArray<BinderInvocation>)> inputData = compilationData.Combine(inputCalls.Collect());
+                    CandidateBinderInvocation.Create)
+                .Where(invocation => invocation is not null)
```

Comment:
It would seem like the initial `IncrementalValuesProvider` node would contain as many elements as there are invocation expressions in the current compilation, with substantial filtering happening in this `Where` clause?

I wonder if there are any performance implications with incremental values holding so many values. Is there any way we could make the predicate more precise (e.g. by peeking at the method name being invoked?)

cc @chsienki @CyrusNajmabadi

Reply:
I haven't looked at the pr at all, but the following is likely very bad: 

> would contain as many elements as there are invocation expressions in the current compilation, 

This will be enormous, and will likely be recomputed on every edit.  Almost certainly not ok. 

I can look more deeply at the actual pr tomorrow.

Reply:
The list is pre-filtered using name matches and should only contain invocations that are very like the binding calls we  care about -https://github.com/dotnet/runtime/blob/d5b41307d26cfe28b05354ea4bb1eba2a7eea27f/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Parser/CandidateBinderInvocation.cs#L18-L20.

I should be able to remove the `Where` clause.

Reply:
Even if the `CandidateBinderInvocation.Create` correctly detects that and returns null, it still results in a very large set of incremental values that is mostly nulls. Making the filtering predicate in line 36 more precise would go a long way in reducing the number of elements in that initial incremental value.

Reply:
Ah yes.